New healthcheck

bludnic opened the pull request:
- [x] Added unified Healthcheck for all nodes (ADM, BTC, ETH, DOGE, DASH, LSK)
- [x] Show additional info for "Unsupported status": `Http is not allowed`, `Outdated API version`, `Not reachable`

![image](https://github.com/Adamant-im/adamant-im/assets/25831507/c2ac439f-9b6d-44ea-a406-d3e6f1569181)

bludnic commented on src/lib/nodes/abstract.node.ts:
```diff
@@ -0,0 +1,153 @@
+  /**
+   * WebSocket protocol
+   */
+  wsProtocol: 'ws:' | 'wss:' = 'wss:'
```
This code `new URL('wss://hello.com').protocol` will return `wss:`. You can see a similar result with `https:` in the constructor.

bludnic commented on src/components/NodesTable/NodesTableItem.vue:
```diff
@@ -11,35 +11,40 @@
+        <v-icon
+          :class="{
+            [classes.statusIcon]: true,
+            [classes.statusIconGreen]: nodeStatusColor === 'green',
```
We don't use computed string for class names

bludnic commented on src/lib/nodes/abstract.client.ts:
```diff
@@ -0,0 +1,119 @@
+import { filterSyncedNodes } from './utils/filterSyncedNodes.ts'
+import { Node } from './abstract.node'
```
removed `.ts` extenstion

bludnic commented on src/lib/nodes/adm/AdmNode.ts:
```diff
@@ -0,0 +1,123 @@
+   * accepts `ApiNode` as a first argument and returns an object.
+   */
+  request<P extends Payload = Payload>(cfg: RequestConfig<P>) {
+    const { url, method = 'get', payload } = cfg
```
I think it's ok to destruct in function body

bludnic commented on src/lib/nodes/btc/BtcNode.ts:
```diff
@@ -0,0 +1,36 @@
+    super(url)
+
+    this.client = axios.create({ baseURL: url })
+    this.client.interceptors.response.use(null, (error) => {
```
added `createBtcLikeClient` helper

bludnic commented on src/store/modules/nodes/nodes-plugin.js:
```diff
@@ -1,22 +1,45 @@
+
+export default (store) => {
+  // initial nodes state
+  adm.getNodes().forEach((status) => store.commit('nodes/status', { status, nodeType: 'adm' }))
```
To iterate all nodes in one place we would need something like `NodesManager` class. If we had 100 nodes it make sense. The current approach is ok. There is no need in another abstraction.

bludnic replied:
I use both named, and default export. See Vuex crypto modules

bludnic commented on src/store/modules/nodes/nodes-actions.js:
```diff
@@ -1,21 +1,31 @@
+  },
+
+  updateStatus() {
+    adm.checkHealth()
```
fixed

bludnic commented:
@adamant-al check the UI for coin nodes. Does it look ok?

adamant-al commented:
1. Add infoService health check
2. Check `alt_ip` if web domain is not accessible
3. Rename `Nodes list` to `ADM nodes`
4. Move coin nodes to the `Coin nodes` and place it after `ADM nodes` in the Settings
5. Add the `Service nodes` for infoService and place it after `Coin nodes` in the Settings
6. Change the padding

<img width="803" alt="Screenshot 2023-11-05 в 17 19 00" src="https://github.com/Adamant-im/adamant-im/assets/33592982/7a4168f5-b217-4d74-a84b-ea7a68c490d9">

7. Remove `Socket` column in the `Coin nodes` screen
8. Move coin name suffix to a separate, the first column, same text format
9. Suffixes: eth-node, eth-indexer, btc-node, btc-indexer, doge-node, doge-indexer, dash-node, lsk-node, lsk-indexer, rates-info. Consider to include them in the `adamant-wallets` repo.
10. Sort by suffix ascending
11. Show coin node version where available
12. Under the ping value, show node's height
13. For infoService, it's `last_update` timestamp — to be added to https://info.adamant.im/get. Show as '13 minutes ago'. Threshold is 30 min (set a constant).
14. Add health check for indexers
15. Later we'll allow disabling coin nodes (not to expose user IP)

bludnic commented:
Created a task https://trello.com/c/Uhj8LOdD/456-healthcheck-improvements

bludnic commented:
@adamant-al fixed "6. Change the padding"

![image](https://github.com/Adamant-im/adamant-im/assets/25831507/99d51b37-86d4-44a3-b89a-529185abfb0d)

For everything else I created a task for future release: https://trello.com/c/Uhj8LOdD/456-healthcheck-improvements

bludnic commented:
@adamant-al need your approval

bludnic commented:
I will fix in the current PR: 3, 4, 6, 7, 8, 9, 10, 11, 12.
Other changes related to indexers/services healthcheck will be done in https://trello.com/c/Uhj8LOdD/456-healthcheck-improvements

bludnic commented:
![image](https://github.com/Adamant-im/adamant-im/assets/25831507/c35dbc4a-e2c0-4d2a-b0b9-0bed63e1a646)
![image](https://github.com/Adamant-im/adamant-im/assets/25831507/e83a7e06-a7c0-4200-9291-2758a218ac5d)

bludnic commented:
- Made `NodesTable` reusable. Decomposed into smaller components
- Split `NodesTable` into `AdmNodeTable` and `CoinNodesTable`
- Added label (suffix) to the Node instance

bludnic commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Feat: Health check for coin nodes and infoservices](https://trello.com/c/MQVhzbjO/232-feat-health-check-for-coin-nodes-and-infoservices)

bludnic commented:
Seems like node's height takes up too much space. Proposed solution:

![image](https://github.com/Adamant-im/adamant-im/assets/25831507/cb24013c-b4b5-4a97-ac74-70415aa5b033)

Icon: https://pictogrammers.com/library/mdi/icon/cube-outline/

@adamant-al

adamant-al commented:
Added the `Format the Nodes screen (screenshots)` list item in the Feat: Health check improvements https://trello.com/c/Uhj8LOdD
You can do a part here.
No tooltips.

bludnic commented:
- Added cube-icon before node height
- Fixed conflicts

RealGoodProgrammer commented on src/components/nodes/adm/AdmNodesTable.vue:
```diff
@@ -0,0 +1,51 @@
+<style lang="scss">
+@import 'vuetify/settings';
+
+.adm-nodes-table {
```
Unnecessary
